helmfile/charts/opensearch/configurer/files/configurer.sh

```diff
@@ -59,6 +59,13 @@ setup_dashboards() {
   fi
 }
 
+check_s3_repository_registered() {
+  resp=$(curl -u "${auth}" --insecure --silent "${os_url}/_snapshot/${snapshot_repository}")
+  if [ "$(echo $resp | jq -r .error.type)" = "repository_missing_exception" ]; then
+    return 1
+  fi
+}
+
 register_s3_repository() {
   echo
   echo "Registering S3 snapshot repository"
```

Comment:
```suggestion
  echo "Registering S3 snapshot repository"
  if $(check_s3_repository_registered); then 
  		echo "Repository is already registered" 1>&2
    	return
 fi
```

Reply:
And then drop the rest, becuase you do not really need to know the `acknowledged` to know if it is registered right?